### PR TITLE
Add policy for deleting CPC repo content

### LINKED ISSUE #1346: Decide on a method to archive content that is no longer current
In #1336, @mhdawson [expressed concern](https://github.com/openjs-foundation/cross-project-council/pull/1336#issuecomment-2234390875) around losing historical context and intent when archiving content that is no longer relevant to the CPC's day to day. This concern was echoed by others in the following CPC call.

I suggest we define a good strategy to archive such content and document it so it is easier to keep the CPC's repo lean.

Here are a few options to get us started.

1. An archive index page linking to permalinks of archived content
2. An archive directory with just a README explaining what it is
3. An archive directory with an archive index

### PR DESCRIPTION
**Pull request by tobie**
Closes #1346.

### CODE REVIEW

File: governance/GOVERNANCE.md [hunk 1]
```diff
@@ -166,6 +166,12 @@ The pull request may be fast-tracked if two CPC members approve the fast-trackin
 
 CPC members may request fast-tracking of pull requests they did not author. In that case only, the request itself is also one fast-track approval. Upvote the comment anyway to avoid any doubt.
 
+### Removing obsolete content
+
+The CPC aims to keep this repository up to date and void of obsolte information. The default is therefore to remove obsolete information. It is always possible to open a pull request to re-include deleted information. Please explain the rationale for doing so in that case, and please make sure such information is clearly labeled as obsolete.
```
Comment (ctcpip) [inline governance/GOVERNANCE.md:171]:
```suggestion
The CPC aims to keep this repository up-to-date and devoid of obsolete information. The default is therefore to remove obsolete information. It is always possible to open a pull request to restore deleted information. Please explain the rationale for doing so in that case, and ensure such information is clearly labeled as obsolete.
```

File: governance/GOVERNANCE.md [hunk 2]
```diff
@@ -166,6 +166,12 @@ The pull request may be fast-tracked if two CPC members approve the fast-trackin
...
+
+In order to make it easier to search for deleted information in the website, it is recommended to pay particular attention to commit messages when deleting content. In particular, note what content is being removed and why.
```
Comment (ctcpip) [inline governance/GOVERNANCE.md:173]:
```suggestion
To facilitate locating deleted information, please pay close attention to commit messages when deleting content. In particular, note what content is being removed and why.
```

**Review by ctcpip — CHANGES_REQUESTED** [review]
minor suggestions for clarity, grammar, spelling

**Review by ctcpip — APPROVED** [review]

**Review by mertcanaltin — APPROVED** [review]
lgtm

**Review by sheplu — APPROVED** [review]
LGTM

**Review by UlisesGascon — APPROVED** [review]

**Pull request merged**